src/transformers/modeling_utils.py

```diff
@@ -1958,7 +1973,8 @@ def _get_resized_lm_head(
         if new_num_tokens is None:
             return old_lm_head
 
-        if is_deepspeed_zero3_enabled():
+        is_quantized = hasattr(self, "hf_quantizer") and self.hf_quantizer is not None
```

Comment:
Is there any reason we couldn't have a `is_quantized` property of the model, which is by default `False`? Having to constantly define `is_quantized` within the methods isn't ideal, as it requires updating in many different places if the criteria change

Reply:
we already have property `hf_quantizer` for the model, storing `is_quantized ` would duplicate the same information. Earlier, I was directly using `self.hf_quantizer is None` in checks but there were comments above to improve readability using `is_quantized`, and as such, I made the changes accordingly.